providers/summary/clearlydefined.js

```diff
@@ -1,15 +1,14 @@
 // Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
 // SPDX-License-Identifier: MIT
 
-const { get, set, isArray, uniq, cloneDeep } = require('lodash')
+const { get, set, isArray, cloneDeep } = require('lodash')
 const SPDX = require('../../lib/spdx')
 const {
   extractDate,
   setIfValue,
   extractLicenseFromLicenseUrl,
   buildSourceUrl,
   updateSourceLocation,
-  isLicenseFile,
   mergeDefinitions
```

Comment:
mergeDefinitions needs to be updated since files is not flat anymore.

When the files get merged, we need to merge the hashes and the natures as well

Reply:
when we update mergeDefintions - make sure NOASSERTION doesn't overwrite something that is set and always gets overwritten by a valid value